xwiki-platform-core/xwiki-platform-refactoring/xwiki-platform-refactoring-api/src/main/java/org/xwiki/refactoring/job/DeleteRequest.java

```diff
@@ -78,21 +81,21 @@ public void setShouldSkipRecycleBin(boolean shouldSkipRecycleBin)
     }
 
     /**
-     * @return the document to be used as the new target after the delete
+     * @return the document to be used as the new backlink target after delete
```

Comment:
There's a problem here. ``DeleteRequest`` extends ``EntityRequest`` which means it targets **multiple** entities. Which is why we have ``getEntityReferences()`` ("the collection of entity references that are targeted by this request"). I think you focused too much on the UI where we delete a single page at once, but at the API level, the delete request can target multiple pages. So I'm not sure it makes sense to have a **single** new backlink target to use for all the deleted pages. It makes more sense to me to have a map ``(deleted entity reference, new backlink reference)``. This way, when you delete multiple pages at once (in the same job, with the same request) you can specify which new backlink reference to use for each of the deleted pages. Moreover, this mapping could be useful also to specify a new backlink target for some of the affected child pages (of course when the delete is done programmatically).

Reply:
So for the UI action this would mean that only the parent document will have a new redirect when selecting `affectChildren`. Should this be better highlighted on the backlinks pane?